WordPress/Classes/Utility/CoreDataHelper.swift

```
@@ -0,0 +1,149 @@
+        } catch {
+            DDLogSwift.logError("Error deleting entity [\(T.entityName)]")
+        }
+    }
+
+    /// Deletes all of the NSMO instances associated to the current kind
+    ///
+    func deleteAllObjects() {
+        let request = newFetchRequest()
+        request.includesPropertyValues = false
+        request.includesSubentities = false
+
+        do {
+            let objects = loadObjects(withFetchRequest: request)
+            for object in objects  {
+                context.deleteObject(object)
+            }
+
+            try context.save()
+        } catch {
+            DDLogSwift.logError("Error deleting all entities of kind [\(T.entityName)]")
+        }
+    }
+
+    /// Retrieves the first entity that matches with a given predicate
+    ///
+    /// - Parameter predicate: Defines the conditions that any given object should meet.
+    ///
+    func firstObject(matchingPredicate predicate: NSPredicate) -> T? {
+        let request = newFetchRequest()
+        request.predicate = predicate
+        request.fetchLimit = 1
+
+        let objects = loadObjects(withFetchRequest: request)
+        return objects.first
+    }
+
+    /// Inserts a new Entity. For performance reasons, this helper *DOES NOT* persists the context.
+    ///
+    func insertNewObject() -> T {
+        let name = T.entityName
+        let entity = NSEntityDescription.insertNewObjectForEntityForName(name, inManagedObjectContext: context)
+
+        return entity as! T
+    }
+
+    /// Loads a single NSManagedObject instance, given its ObjectID, if available.
+    ///
+    /// - Parameter objectID: Unique Identifier of the entity to retrieve, if available.
+    ///
+    func loadObject(withObjectID objectID: NSManagedObjectID) -> T? {
+        var result: T?
+
+        do {
+            result = try context.existingObjectWithID(objectID) as? T
+        } catch {
+            DDLogSwift.logError("Error loading Object [\(T.entityName)]")
```

Silent error failure.